### PR TITLE
[SPARK-13594][SQL] remove typed operations(e.g. map, flatMap) from python DataFrame

### PR DESCRIPTION
**Pull request by cloud-fan**
## What changes were proposed in this pull request?

Remove `map`, `flatMap`, `mapPartitions` from python DataFrame, to prepare for Dataset API in the future.
## How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @rxin @yhuai

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -267,44 +267,6 @@ def take(self, num):
...
-
-        This is a shorthand for ``df.rdd.mapPartitions()``.
-
-        >>> rdd = sc.parallelize([1, 2, 3, 4], 4)
-        >>> def f(iterator): yield 1
-        >>> rdd.mapPartitions(f).sum()
-        4
-        """
-        return self.rdd.mapPartitions(f, preservesPartitioning)
-
```
Comment (yhuai) [inline python/pyspark/sql/dataframe.py]: Should we also remove `foreach` and `foreachPartition`?

Comment (rxin) [inline python/pyspark/sql/dataframe.py]: those are fine, since they don't return anything.

**Comment by rxin** [comment]
Thanks - merging this in master.

**Comment by mydpy** [comment]
This change surprised me as a user of Pyspark on the 2.0.0-Snapshot. Thanks for documenting this well. Since I usually use the Scala API, it was not clear to me that Pyspark didn't support the Datasets API yet (i.e., `df.rdd.flatMap(...)` returns a PythonRDD as-opposed to a Dataset)

**Comment by maver1ck** [comment]
@rxin 
As we're not planning to implement DataSets in Python is there a plan to revert this Jira ?

**Pull request closed without merging**